File: javascript/queries/scheduleRequests.js
const { gql } = require("graphql-request")
const client = require("../utils/graphqlClient")

const query = gql`
  query ScheduleRequests($inputs: ScheduleRequestsInputs!){
    scheduleRequests(inputs: $inputs){
      pagination {
        totalCount
        currentPage
      }
      scheduleRequests{
        id
        status
        order {
          status
        }
      }
    }
  }
`

client.request(
  query,
  { inputs: { status: "OPEN" } }
).then((data) => console.log(data))

File: ruby/queries/schedule_requests.rb
require_relative "../utils/graphql_client"

ScheduleRequestsQuery = Client.parse <<-'GRAPHQL'
  query($inputs: ScheduleRequestsInputs!){
    scheduleRequests(inputs: $inputs){
      pagination {
        totalCount
        currentPage
      }
      scheduleRequests{
        id
        status
        externalReferenceId
        order {
          status
        }
      }
    }
  }
GRAPHQL

result = Client.query(
  ScheduleRequestsQuery,
  variables: { inputs: { status: "OPEN" } }
)
pp result.data.to_h
